packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogLogEventMapper.kt

```diff
@@ -0,0 +1,113 @@
+package com.datadoghq.flutter
+
+import android.os.Handler
+import android.os.Looper
+import com.datadog.android.Datadog
+import com.datadog.android.log.LogsConfiguration
+import com.datadog.android.log.model.LogEvent
+import io.flutter.plugin.common.MethodChannel
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+/**
+ * This is a helper class that that simplifies event mapping / scrubbing for Log events.
+ *
+ * Since it is possible for Flutter engines to shut down and be recreated, it is possible that the
+ * object used as the event mapper to be lost. This class is used as a static instance that will
+ * perform mapping and call the Flutter mapping methods using the last provided MethodChannel.
+ */
+internal class DatadogLogEventMapper {
+    private val channels: MutableSet<MethodChannel> = mutableSetOf()
```

Comment:
also the thread-safe access collection should be used here if it is possible to access it from the different threads